File: Front-end/store/src/app/core/models/product-preview.ts
import { Image } from '../interfaces/image';

export class ProductPreview {
  public id: number;

  public name: string;

  public titleImage: Image;

  public cost: number;

  public constructor(productPreview: ProductPreview) {
    this.id = productPreview.id;
    this.name = productPreview.name;
    this.titleImage = productPreview.titleImage;
    this.cost = productPreview.cost;
  }
}

File: Back-end/BookStore.WebApi/Areas/Store/ViewModels/Cards/ProductPreview.cs

using BookStore.Application.Dto;

namespace BookStore.WebApi.Areas.Store.ViewModels.Cards
{
    public abstract record ProductPreview
    {
        public int Id { init; get; }
        public string Name { init; get; }
        public int Cost { init; get; }
        public ImageDto TitleImage { init; get; }
    }
   
}
